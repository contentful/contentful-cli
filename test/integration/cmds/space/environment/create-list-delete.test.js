const nixt = require('nixt')
const { join } = require('path')
const { initConfig, deleteSpaces, createSimpleSpace } = require('../../../util')

const bin = join(__dirname, './../../../../../', 'bin')

const app = () => {
  return nixt({ newlines: true })
    .cwd(bin)
    .base('./contentful.js ')
    .clone()
}

const org = process.env.CLI_E2E_ORG_ID
let space = null
const spacesToDelete = []

beforeAll(async () => {
  await initConfig()
  space = await createSimpleSpace(org, 'space-env')
  spacesToDelete.push(space.sys.id)
})

afterAll(() => {
  return deleteSpaces(spacesToDelete)
})

test('should create, list and delete environment', done => {
  function createEnvironment() {
    app()
      .run(
        `space environment create --space-id ${space.sys.id} --environment-id createListDelete --name "Create List Delete"`
      )
      .expect(result => {
        const resultText = result.stdout.trim()
        expect(resultText).toMatchSnapshot()
      })
      .code(0)
      .end(listEnvironments)
  }

  function listEnvironments() {
    app()
      .run(`space environment list --space-id ${space.sys.id}`)
      .stdout(/Environment name +|/)
      .stdout(/Environment id +|/)
      .stdout(/Create List Delete +|/)
      .stdout(/master +|/)
      .code(0)
      .end(deleteEnvironment)
  }

  function deleteEnvironment() {
    app()
      .run(
        `space environment delete --space-id ${space.sys.id} --environment-id createListDelete`
      )
      .stdout(/Environment name +|/)
      .stdout(/Environment id +|/)
      .stdout(/Create List Delete +|/)
      .stdout(/master +|/)
      .code(0)
      .end(done)
  }

  createEnvironment()
}, 10000)
