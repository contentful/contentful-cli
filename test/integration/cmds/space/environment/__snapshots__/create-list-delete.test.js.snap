// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create, list and delete environment 1`] = `"Successfully created environment Create List Delete (createListDelete)"`;
